src/main/java/com/alex/eshop/eshop/Repository/CategoryRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.alex.eshop.eshop.Repository;
+
+import com.alex.eshop.eshop.Entity.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface CategoryRepository extends JpaRepository<Category, Integer> {
+    Optional<Category> findById(Long id);
```

Comment:
Function `Optional<Category> findById(Long id);` is redundant, because `interface CrudRepository` already have this function. You cannot use it because you defined wrong ID Type in `JpaRepository<Category, Integer>`. It's not Integer.